### PR TITLE
improve createDeepProxy docs

### PR DESCRIPTION
**Pull request by YPAzevedo**
This PR has some suggestion on docs improvements for `createDeepProxy`

### CODE REVIEW

**Comment by dai-shi** [comment]
Thanks for the suggestion. Could you modify the jsdoc in the code? API section in README is auto generated.

File: README.md [hunk 1]
```diff
@@ -54,22 +54,31 @@ true
...
+import { createDeepProxy } from "proxy-compare";
+
+const orginal = { a: "1", c: "2", d: { e: "3" } };
+const affecteProps = new WeakMap();
+
+const proxy = createDeepProxy(orginal, affecteProps);
+
+proxy.a // Will mark as used and track its value.
+// This will update the affected WeakMap with orginal as key
+// and a Set with "a"
```
Comment (dai-shi) [inline README.md:77]: It's really hard to look inside the affected WeakMap. For experiment, you can specify a Map instead and which is printable. We also have `affectedToPathList` for printing.

Comment (YPAzevedo) [reply README.md:77]: Should we briefly  say on the example that for printing the WeakMap they should use  `affectedToPathList`  we can elaborate more on it in another PR when adding the documentation for `affectedToPathList`. What do you think?

File: README.md [hunk 2]
```diff
@@ -54,25 +54,34 @@ true
...
+
+proxy.d // Will mark "d" as accessed to track and proxy itself ({ e: "3" }).
+// This will update the affected WeakMap with orginal as key
+// and a Set with "d"
 ```
 
-Returns **T** 
+Returns **[object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Object)** Object wrapped in a proxy..
```
Comment (YPAzevedo) [inline README.md:84]: Wouldn’t this be `Proxy<object>` ?

Comment (dai-shi) [reply README.md:84]: yeah, previously, documentationjs just picks type information. Now, you added params manually in the jsdoc, we can change it.

File: README.md (hunk 1 above)
Comment (dai-shi) [reply README.md:77]: sounds good.

**Comment by dai-shi** [comment]
Thanks for the change.
Please revert the change in README.md and focus on changes/discussions in index.ts.

File: src/index.ts [hunk 3]
```diff
@@ -120,16 +120,35 @@ const createProxyHandler = <T extends object>(origObj: T, frozen: boolean) => {
...
+ * @returns {Proxy<object>} - Object wrapped in a proxy.
  *
  * @example
  * import { createDeepProxy } from 'proxy-compare';
  *
- * const obj = ...;
- * const affected = new WeakMap();
- * const proxy = createDeepProxy(obj, affected);
+ * const orginal = { a: "1", c: "2", d: { e: "3" } };
+ * const affecteProps = new WeakMap();
```
Comment (dai-shi) [inline src/index.ts:142]: do you mean `affectedProps`? or it is the word i never used..

btw, it's not technically props it's `<obj, props>`

**Review by dai-shi — COMMENTED** [review]
looks good. just one comment.

File: src/index.ts (hunk 3 above)
Comment (YPAzevedo) [reply src/index.ts:142]: yeah `affecteProps` it was a typo. ill update the const name to `affected` to avoid any confusion 💯

Comment (YPAzevedo) [reply src/index.ts]: done

**Review by dai-shi — APPROVED** [review]
Thanks. Let me merge this. I will update README on master.

**Pull request merged**